components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -1311,6 +1321,16 @@ private String handleFailureAuthorization(OAuthMessage oAuthMessage, OIDCSession
         return EndpointUtil.getErrorRedirectURL(oAuthMessage.getRequest(), oauthProblemException, oauth2Params);
     }
 
+    private String handleFailureBeforeConsent(OAuthMessage oAuthMessage, OAuth2Parameters oauth2Params,
+                                              OAuth2AuthorizeRespDTO authzRespDTO) {
```

Comment:
Shall we use a better name for this method, like `handleAuthorizationFailureBeforeConsent` as it would be better if you add which failure is going to be handled from this method.